crates/proof-of-sql/src/base/proof/transcript_core.rs

```diff
@@ -15,6 +15,9 @@ pub(super) trait TranscriptCore {
 }
 
 /// private method to facilitate recieving challenges and reversing them. Undefined behavior if the size_of `M` is not 32 bytes.
+///
+/// # Panics
+/// - Panics if `M::read_from(&bytes)` fails to read the bytes into the type `M`.
```

Comment:
Can this happen? What would cause `read_from` to fail?

Reply:
ig yes